### PR TITLE
Finally... remove Point

### PR DESCRIPTION
**Pull request by cnr12**
Fixing the width of geometry to 3, this allows replacing `geometry::Point` with `Eigen::Vector3d` everywhere. Apologies for the huge diff, but `Point` was everywhere...
Also, sorry for creating this as "cnr12", I was logged in on another account.

### CODE REVIEW

File: cpp/dolfin/mesh/IntervalCell.cpp [hunk 1]
```diff
@@ -170,12 +170,13 @@ geometry::Point IntervalCell::cell_normal(const Cell& cell) const
...
-  geometry::Point p1 = geometry.point(vertices[1]);
+  Eigen::Vector3d p0 = geometry.x(vertices[0]);
+  Eigen::Vector3d p1 = geometry.x(vertices[1]);
 
   // Define normal by rotating tangent counterclockwise
-  geometry::Point t = p1 - p0;
-  geometry::Point n(-t[1], t[0]);
+  Eigen::Vector3d t = p1 - p0;
+  Eigen::Vector3d n;
+  n << -t[1], t[0], 0.0;
```
Comment (michalhabera) [inline cpp/dolfin/mesh/IntervalCell.cpp]: RAII

File: cpp/demo/hyperelasticity/main.cpp [hunk 2]
```diff
@@ -164,8 +164,10 @@ int main(int argc, char* argv[])
   // .. code-block:: cpp
 
   // Create mesh and define function space
-  std::array<geometry::Point, 2> pt
-      = {geometry::Point(0.0, 0.0, 0.0), geometry::Point(1.0, 1.0, 1.0)};
+  std::array<Eigen::Vector3d, 2> pt;
```
Comment (garth-wells) [inline cpp/demo/hyperelasticity/main.cpp]: Does `std::array<Eigen::Vector3d, 2> pt{(0, 0, 0), (1, 1, 1)};` work?

Comment (chrisrichardson) [reply cpp/demo/hyperelasticity/main.cpp]: Unfortunately not. Very little works for Eigen initialisation.

File: cpp/dolfin/geometry/CollisionPredicates.cpp [hunk 3]
```diff
@@ -28,15 +29,17 @@ static inline Point cross_product(const Point& a, const Point& b,
   double axy[2] = {a[0], a[1]};
   double bxy[2] = {b[0], b[1]};
   double cxy[2] = {c[0], c[1]};
-  return Point(_orient2d(ayz, byz, cyz), _orient2d(azx, bzx, czx),
-               _orient2d(axy, bxy, cxy));
+  Eigen::Vector3d r;
+  r << _orient2d(ayz, byz, cyz), _orient2d(azx, bzx, czx),
```
[the remark this answers is not in the document]
Comment (chrisrichardson) [reply cpp/dolfin/geometry/CollisionPredicates.cpp]: Have you tried it?

[the remark this answers is not in the document]
Comment (michalhabera) [reply cpp/dolfin/geometry/CollisionPredicates.cpp]: No, but its in documentation, https://eigen.tuxfamily.org/dox/group__QuickRefPage.html#title1

File: cpp/dolfin/geometry/BoundingBoxTree.h [hunk 4]
```diff
@@ -251,7 +250,7 @@ class BoundingBoxTree
 
   // Compute bounding box of points
   static void compute_bbox_of_points(
-      double* bbox, std::size_t& axis, const std::vector<Point>& points,
+                                     double* bbox, std::size_t& axis, const std::vector<Eigen::Vector3d>& points,
```
[the remark this answers is not in the document]
Comment (chrisrichardson) [reply cpp/dolfin/geometry/BoundingBoxTree.h]: Yes, I probably didn't run clang-format on the header files (only cpp)... will try to fix.

File: cpp/dolfin/geometry/CollisionPredicates.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (chrisrichardson) [reply cpp/dolfin/geometry/CollisionPredicates.cpp]: OK, thanks. Perhaps more works than I thought...

File: cpp/demo/hyperelasticity/main.cpp (hunk 2 above)
Comment (chrisrichardson) [reply cpp/demo/hyperelasticity/main.cpp]: `std::array<Eigen::Vector3d, 2> pt{Eigen::Vector3d(0.0, 0.0, 0.0),
                                    Eigen::Vector3d(1.0, 1.0, 0.0)};`. How about that?

**Pull request merged**